### PR TITLE
Issue #DP-662 feat: open for contributions

### PR DESCRIPTION
**Pull request by venkateshwarans**

### CODE REVIEW

File: src/app/client/src/app/modules/cbse-program/components/recursive-tree/recursive-tree.component.html [hunk 1]
```diff
@@ -5,12 +5,15 @@
...
   [telemetryInteractPdata]="telemetryInteractPdata"
   [isOpen]=" (collection.identifier === sessionContext.lastOpenedUnitChild || collection.identifier === sessionContext.lastOpenedUnitParent) ? true : false">
-    <div title class="sb-accordion-title d-flex flex-jc-flex-end sb-color-primary"><span class="chapter-status mr-16"></span>
+    <div title class="sb-accordion-title d-flex flex-jc-flex-end sb-color-primary"><span class="chapter-status mr-16 mt-5"></span>
      <span class="flex-14"> {{collection.name}}</span>
       <div class="sb-color-black flex-3">
         <span class="font-weight-normal">{{resourceService.frmelmnts.lbl.totalContent}}:</span> <span class="font-weight-bold"> {{collection.totalLeaf}}</span>
       </div>
      <div class="d-flex">
+        <button class="sb-btn sb-btn-xs sb-btn-outline-primary sb-left-icon-btn library-btn mr-8"
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/cbse-program/components/recursive-tree/recursive-tree.component.html:14]: what is this button for? 
@venkateshwarans

File: src/app/client/src/app/modules/program/components/list-contributor-textbooks/list-contributor-textbooks.component.html [hunk 2]
```diff
@@ -15,102 +15,102 @@
...
+																<br />
+																<em [class.sb-color-warning]="activeDate === 'enddate'"
+																	*ngIf="activeDate === 'enddate'">{{ programDetails?.enddate | daysToGo  }}</em>
+															</div>
+														</div>
+													</div>
+												</div>
+											</ng-container>
+										</div>
 		</div>
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/list-contributor-textbooks/list-contributor-textbooks.component.html:114]: The nominations Tab should not be shown 
@venkateshwarans

File: src/app/client/src/app/modules/program/components/create-program/create-program.component.ts [hunk 3]
```diff
@@ -463,7 +464,19 @@ export class CreateProgramComponent implements OnInit, AfterViewInit {
...
   }
-
+  openForNominations(status) {
+    this.isOpenNominations = status;
+    if (status) {
+      this.createProgramForm.controls['nomination_enddate'].setValidators(Validators.required);
+    } else {
+    this.createProgramForm.controls['nomination_enddate'].clearValidators();
+    this.createProgramForm.controls['shortlisting_enddate'].clearValidators();
+    this.createProgramForm.controls['nomination_enddate'].setValue(null);
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:474]: the values should not be set to null here. As if I check, uncheck and again check the box the values should be there. While saving the the check should be there.

File: src/app/resourcebundles/data/consumption/en.properties [hunk 4]
```diff
@@ -1401,6 +1402,9 @@ frmelmnts.lbl.nominationDescription=Are you sure you want to submit this nominat
 frmelmnts.lbl.submitNomination=Submit Nomination
 #MyProject user tab
 frmelmnts.lbl.assignUsersToProject=Assign users to project
+frmelmnts.lbl.yes = Yes
+frmelmnts.lbl.no = No
```
Comment (twsvaishali) [inline src/app/resourcebundles/data/consumption/en.properties:1406]: frmelmnts.btn.yes and frmelmnts.btn.no are already there. Can we reuse that?

File: src/app/client/src/app/modules/cbse-program/components/recursive-tree/recursive-tree.component.html (hunk 1 above)
Comment (venkateshwarans) [reply src/app/client/src/app/modules/cbse-program/components/recursive-tree/recursive-tree.component.html:14]: This is related to Content Reuse. where we are adding content from MVC Library

File: src/app/client/src/app/modules/program/components/list-contributor-textbooks/list-contributor-textbooks.component.html (hunk 2 above)
Comment (venkateshwarans) [reply src/app/client/src/app/modules/program/components/list-contributor-textbooks/list-contributor-textbooks.component.html:114]: That's okay I guess. Sourcing Org will see his own nomination only. But PM can take a call later.

Comment (twsvaishali) [reply src/app/client/src/app/modules/program/components/list-contributor-textbooks/list-contributor-textbooks.component.html:114]: In the demo today they said. that is why I remembered. :)

Comment (venkateshwarans) [reply src/app/client/src/app/modules/program/components/list-contributor-textbooks/list-contributor-textbooks.component.html:114]: Okay. Santosh is already working on that. will send a separate PR

File: src/app/client/src/app/modules/cbse-program/components/recursive-tree/recursive-tree.component.html [hunk 5]
```diff
@@ -5,12 +5,15 @@
...
+        *ngIf="visibility && visibility['showAddresource'] && programContext?.status === 'Unlisted'">
+          <i class="add icon mr-4"></i>{{resourceService.frmelmnts.lbl.addToLibrary}}</button>
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/cbse-program/components/recursive-tree/recursive-tree.component.html:16]: @venkateshwarans  programContext?.status === 'Unlisted'" should be  programContext?.type === 'Private'"

File: src/app/client/src/app/modules/contribute/components/list-nominated-textbooks/list-nominated-textbooks.component.html [hunk 6]
```diff
@@ -1,119 +1,118 @@
...
+							<div class="status" [class.active]="activeDate === 'nomination_enddate'">
+							  <div class="status__label">{{resourceService.frmelmnts.lbl.nominationEndDate}}</div>
+							  <div class="status__date">
+								{{ programDetails?.nomination_enddate |  date:'LLLL d, yyyy'  }}
+								<br/>
+								<em [class.sb-color-warning]="activeDate === 'nomination_enddate'" *ngIf="activeDate === 'nomination_enddate'">{{ programDetails?.nomination_enddate | daysToGo  }}</em>
+							  </div>
+							</div>
+						  </div>
+						  <div class="program-timeline-item" *ngIf="programDetails?.status !== 'Unlisted'">
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/contribute/components/list-nominated-textbooks/list-nominated-textbooks.component.html:71]: programContext?.status === 'Unlisted'" should be  programContext?.type === 'Private'"

File: src/app/client/src/app/modules/contribute/components/list-nominated-textbooks/list-nominated-textbooks.component.html [hunk 7]
```diff
@@ -1,119 +1,118 @@
...
+							  <div class="status__date">
+								{{ programDetails?.enddate |  date:'LLLL d, yyyy'  }}
+								<br/>
+								<em [class.sb-color-warning]="activeDate === 'enddate'" *ngIf="activeDate === 'enddate'">{{ programDetails?.enddate | daysToGo  }}</em>
+							  </div>
+							</div>
+						  </div>
+						</div>
+					  </ng-container>
+					<div class="sb-meta-data-container sb-meta-data-column px-10" *ngIf="programDetails?.status !== 'Unlisted'">
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/contribute/components/list-nominated-textbooks/list-nominated-textbooks.component.html:103]: programContext?.status === 'Unlisted'" should be  programContext?.type === 'Private'"

File: src/app/client/src/app/modules/core/services/programs/programs.service.ts [hunk 8]
```diff
@@ -504,7 +504,7 @@ export class ProgramsService extends DataService implements CanActivate {
     return this.API_URL(req).pipe(tap((res) => {
       if (res.result.program_id) {
         const programId = res.result.program_id;
-        if (request.status == 'Live') {
+        if (request.status === 'Live' || request.status === 'Unlisted') {
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/core/services/programs/programs.service.ts:507]: request.status === 'Unlisted'" condition we can remove

File: src/app/client/src/app/modules/program/components/create-program/create-program.component.ts [hunk 9]
```diff
@@ -806,7 +821,11 @@ export class CreateProgramComponent implements OnInit, AfterViewInit {
     this.programConfig.medium = this.mediumOption;
     this.programConfig.subject = this.subjectsOption;
     data['config'] = this.programConfig;
-    data['status'] = 'Live';
+    if (this.isOpenNominations) {
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:824]: This code shall be removed

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.html [hunk 10]
```diff
@@ -1,80 +1,80 @@
...
+                            </ng-template>
+                        <p class="fs-0-785 font-weight-normal pr-10 sb__ellipsis sb__ellipsis--three">{{ programDetails?.description }} </p>
+                    </div>
+                    <div class="sb-meta-data-container sb-meta-data-column w-10 px-10 w-30">
+                            <label class="mb-0">{{resourceService?.frmelmnts?.lbl?.contentType}}</label>
+                            <p class="sb__ellipsis sb__ellipsis--one cursor-pointer" suiPopup popupHeader="{{resourceService.frmelmnts.lbl.contentType}}" popupText="{{ programContentTypes }}"><span>{{ programContentTypes }}</span></p>
+                    </div>
+                    <ng-container *ngIf="programDetails">
+                        <div class="program-timeline w-35">
+                            <div class="program-timeline-item" *ngIf="programDetails?.status !== 'Unlisted'">
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.html:31]: programDetails?.status !== 'Unlisted'", here the check should be for programDetails?type

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.html [hunk 11]
```diff
@@ -1,80 +1,80 @@
...
-                    </div>
-                    <div class="program-timeline-item">
-                        <div class="status" [class.active]="activeDate === 'shortlisting_enddate'">
-                            <div class="status__label">{{ resourceService?.frmelmnts?.lbl?.shortlistingNomination }}</div>
-                            <div class="status__date">
-                                {{ programDetails?.shortlisting_enddate |  date:'LLLL d, yyyy'  }}
-                                <br />
-                                <em [class.sb-color-warning]="activeDate === 'shortlisting_enddate'"
-                                    *ngIf="activeDate === 'shortlisting_enddate'">{{ programDetails?.shortlisting_enddate | daysToGo  }}</em>
+                            <div class="program-timeline-item" *ngIf="programDetails?.status !== 'Unlisted'">
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.html:42]: Here as well

File: src/app/client/src/app/modules/shared-feature/components/program-list/program-list.component.html [hunk 12]
```diff
@@ -88,7 +88,7 @@ <h4 class="sb-pageSection-title">
                 <td *ngIf="!isContributor || activeAllProgramsMenu" class="nowrap">
                   <div class="program-timeline">
                     <div class="program-timeline-item">
-                      <div class="status" [class.active]="isActive(program, 'nomination_enddate', i)">
+                      <div class="status" [class.active]="isActive(program, 'nomination_enddate', i)" *ngIf="program.status !== 'Unlisted'">
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/shared-feature/components/program-list/program-list.component.html:91]: Here as well

File: src/app/client/src/app/modules/cbse-program/components/recursive-tree/recursive-tree.component.html (hunk 5 above)
Comment (venkateshwarans) [reply src/app/client/src/app/modules/cbse-program/components/recursive-tree/recursive-tree.component.html:16]: @pallakartheekreddy correct me if I am wrong. The `Add from Library` feature is for projects which are only 'Unlisted' and not based on private or public, right? What if there is a `private` project which is open for nominations?

File: src/app/client/src/app/modules/program/components/create-program/create-program.component.ts (hunk 3 above)
Comment (venkateshwarans) [reply src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:476]: Once merged, I will get it fixed.

**Pull request merged**